nav2_controller/include/nav2_controller/plugins/simple_goal_checker.hpp

```diff
@@ -79,6 +80,8 @@ class SimpleGoalChecker : public nav2_core::GoalChecker
   rclcpp::node_interfaces::OnSetParametersCallbackHandle::SharedPtr dyn_params_handler_;
   std::string plugin_name_;
 
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros_;
```

Comment:
You don't need to store if its not going to be used. In the `cpp` file just change `costmap_ros` to `/*costmap_ros*/` to make the linters happy (so no arguments unused)